src/search_iterator.cpp

```diff
@@ -103,11 +104,10 @@ std::string search_iterator::get_path() const {
     }
 
     std::string path = internal->get_document().get_data();
-    std::string dbData = internal->search->internal->database.get_metadata("data");
+    auto impl = internal->search->m_archives.at(get_fileIndex()).getImpl();
 
-    // zim with old namespace scheme do not have the metadata. In that case, we
-    // should return full path for consistency with Item::getPath and Entry::getPath
-    if(!dbData.empty() && dbData == "fullPath") {
+    // If the archive has new namespace scheme, we return the path without namespace
+    if(impl->hasNewNamespaceScheme()) {
```

Comment:
We need to remove the namespace if `hasNewNamespaceScheme()` is true AND if there is a namespace to remove (`dbData == "fullPath"`).

In the future we will generate index without namespace in it. We want the "current" libzim being ready to handle this.